plugins/discovery/endpoint_creator_test.go

```diff
@@ -0,0 +1,73 @@
+package discovery
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/internal/discovery"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/internal/metrics"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/internal/util"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/discovery/prometheus"
+	"github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/discovery/telegraf"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func makeDummyProviders(handler metrics.ProviderHandler) map[string]discovery.ProviderInfo {
+	providers := make(map[string]discovery.ProviderInfo, 2)
+	providers["prometheus"] = prometheus.NewProviderInfo(handler, "prom.")
+	providers["telegraf"] = telegraf.NewProviderInfo(handler)
+	return providers
+}
+
+func Test_endpointCreator_discoverEndpoints_annotations_dont_happen_when_disabled(t *testing.T) {
+	e := &endpointCreator{
+		delegates:                  nil,
+		providers:                  makeDummyProviders(util.NewDummyProviderHandler(1)),
+		disableAnnotationDiscovery: true,
+	}
+
+	resource := discovery.Resource{
+		Kind:       discovery.PodType.String(),
+		IP:         "0.0.0.0",
+		Meta:       metav1.ObjectMeta{},
+		Containers: make([]v1.Container, 0),
+	}
+
+	got := e.discoverEndpoints(resource)
+	assert.Equal(t, 0, len(got))
+}
+
+func Test_endpointCreator_discoverEndpoints_annotations_happen_when_not_disabled(t *testing.T) {
+	e := &endpointCreator{
+		delegates:                  nil,
+		providers:                  makeDummyProviders(util.NewDummyProviderHandler(1)),
+		disableAnnotationDiscovery: false,
+	}
+
+	resource := discovery.Resource{
+		Kind:       discovery.PodType.String(),
+		IP:         "0.0.0.0",
+		Meta:       metav1.ObjectMeta{},
+		Containers: make([]v1.Container, 0),
+	}
+
+	got := e.discoverEndpoints(resource)
+	assert.Equal(t, 1, len(got))
+}
+
+func Test_endpointCreator_discoverEndpoints_annotations_happen_by_default(t *testing.T) {
+	e := &endpointCreator{
+		delegates: nil,
+		providers: makeDummyProviders(util.NewDummyProviderHandler(1)),
+	}
+
+	resource := discovery.Resource{
+		Kind:       discovery.PodType.String(),
+		IP:         "0.0.0.0",
+		Meta:       metav1.ObjectMeta{},
+		Containers: make([]v1.Container, 0),
```

Comment:
Annotation again. Might just be worth pulling out a createResourceWithAnnotation method, or a setup method.

Reply:
I intentionally omitted it here for this specific test